File: El Dr. Zoidberg No Sabe Operar/El Dr. Zoidberg No Sabe Operar.cpp
#include<stdio.h>
#define Hd main(void)

Hd
{
    int n;
    scanf("%d",&n);
    int a,aux=0;
    for(int i=1;i<=n;i++){
        scanf("%d",&a);
        a+=aux;
        if(a<i||(i==n&&a>n)){
            printf("No");
            return 0;
        }
        aux = a-i;
    }
    printf("Si");
}

File: El Dr. Zoidberg No Sabe Operar/El Dr. Zoidberg No Sabe Operar.py
#define Hd
n : int
n = int(input())
a : int 
aux : int = 0
i : int = 1
for _ in map(int,input().split()):
    a=_
    a+=aux
    if a<i or (i==n and a>n):
        print("No")
        exit(0)
    aux = a-i
    i+=1

print("Si")
